torch/csrc/jit/passes/dtype_analysis.cpp

```diff
@@ -240,8 +290,24 @@ struct DtypePropagationPass {
   bool processAtenOps(Node* n) {
     GRAPH_DEBUG("processAtenOps");
     GRAPH_DEBUG("case = ", n->kind(), " ", *n);
+    // Custom Rule Matching
+    if (auto prop_fn = dtype_prop_registry_->find(n->getOperator())) {
+      DtypePropRule rule = *prop_fn;
+      return rule(n);
+    }
     return tryApplyDtypeMetaTensor(n);
   }
+
+  void buildDtypeRuleRegistry() {
+    // building a registry for all of the custom dtype rules
+    dtype_prop_registry_ = std::make_unique<OperatorMap<DtypePropRule>>();
+
+    dtype_prop_registry_->insert(
```

Comment:
Could we add tests for the new operators that we're using here ? these are still untested